addons/l10n_mx/models/account_account.py

```diff
@@ -17,3 +26,23 @@ def create(self, vals_list):
             tag_id = debit_tag.id if account.code[0] in DEBIT_CODES else credit_tag.id
             account.tag_ids = [Command.link(tag_id)]
         return accounts
+
+    @api.constrains('code', 'group_id')
+    def ensure_code_conforms_coa_sat(self):
+        # filtering accounts used in Mexican reports that do not respect the regulation of the SAT (see documentation for more explanation)
+        # six first characters correspond to the group provided, a list is provided by the SAT
+        # last part is up to the user but should match ACCOUNT_ID_PATTERN which is detailed in the header
+        incorrect_accounts = self.filtered(
+            lambda a: a.group_id and a.account_type != 'equity_unaffected' and a.company_id.country_code == 'MX'
```

Comment:
Note that in 18.0+, accounts can be shared between multiple companies, the code depends on the company, and writing on the code only modifies it for the currently active company.

So, you'll need to:
1. only run the check if the _active_ company is in Mexico, and
2. change the condition so that it checks accounts that belong to the currently active company:

```suggestion
            lambda a: a.group_id and a.account_type != 'equity_unaffected' and self.env.company in a.company_ids
```